### DISCUSSION TITLE
onvisibiltychange does not work for me. help with debugging?

### DISCUSSION THREAD

**Discussion by guystreeter** [cockpit-project/cockpit#17971] [Q&A] [answered]
I want my plugin to stop updating when the user navigates away from it. I have successfully used the `onvisibilitychange` event in a starter-kit test, but I can't make it work in my plugin.
I originally thought the difference might be that I am using functional components instead of classes, but I converted the starter-kit to a functional component implementation, and I catch the event there. I just can't get it to work in my more complex plugin.
I've tried setting it up in the main function code, and I've also tried wrapping it in a `useEffect` so it is set up after the first render, but my function never runs.

**Comment by guystreeter** [comment, marked as answer]
Any suggestions on a solution, or ways to debug? Thanks!

**Reply by jelly** [reply]
Do you have some example code / project link somewhere? The reason why you might have issues is that every plugin lives in an iframe, I am not sure if it receives the `onvisibilitychange` event.

**Reply by guystreeter** [reply]
I have no explanation for it, but it is working now. Thanks for your help!
